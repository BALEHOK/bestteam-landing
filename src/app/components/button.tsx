import classNames from 'classnames';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { colors, fonts, mediaQueries } from '../lib/styleUtils';

type Props = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;

const useStyles = createUseStyles({
  root: {
    alignSelf: 'center',
    background: colors.orange,
    border: 'none',
    borderRadius: 3,
    color: colors.beige,
    fontFamily: fonts.bt,
    fontSize: 32,
    padding: '8px 18px',
    textTransform: 'uppercase',
    [mediaQueries.upXs]: {
      borderRadius: 5,
      fontSize: 48,
      padding: '10px 20px',
    },
    [mediaQueries.upLg]: {
      borderRadius: 10,
      fontSize: 60,
      padding: '12px 26px',
    },

    '&:hover': {
      backgroundColor: colors.orange2,
    },
    '&:active': {
      filter: 'brightness(85%)',
    },
  },
});

export const Button = ({ children, className, ...rest }: Props) => {
  const classes = useStyles();

  return (
    <button className={classNames(classes.root, className)} {...rest}>
      {children}
    </button>
  );
};
