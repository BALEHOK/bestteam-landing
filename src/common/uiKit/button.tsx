import { TFunction } from '../lib/functionTypes';
import { createUseStyles } from 'react-jss';
import classNames from 'classnames';

export enum ButtonType {
    primary = 'primary',
    secondary = 'secondary'
}

export interface Props {
    children: string,
    className?: string
    onClick: TFunction,
    type?: ButtonType,
}

const useStyles = createUseStyles({
  root: {
    height: '58px',
    width: '180px',
    border: 'none',
    textAlign: 'center',
    fontSize: '14px',
    fontWeight: 500,
    borderRadius: '12px',
    transition: 'box-shadow 200ms ease 0s, background 200ms ease 0s',
    boxShadow: 'none',
  },
  primaryButtonStyle: {
    background: 'rgb(17, 20, 45)',
    color: 'rgb(255, 255, 255)',
  },
  secondaryButtonStyle: {
    background: 'rgb(228, 228, 228)',
    color: 'rgb(17, 20, 45)',
  }
});

export const Button = ({ type, children, onClick, className }: Props) => {
  const classes = useStyles();

  let buttonTypeClass: string;
  switch (type) {
    case ButtonType.secondary:
      buttonTypeClass = classes.secondaryButtonStyle;
      break;
    default:
      buttonTypeClass = classes.primaryButtonStyle;
  }

  return (
    <button className={classNames(
      classes.root,
      buttonTypeClass,
      className)}
    onClick={onClick}>{children}</button>
  );
};