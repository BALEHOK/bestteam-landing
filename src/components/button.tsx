import classNames from 'classnames';
import { ReactNode } from 'react';
import { createUseStyles } from 'react-jss';
import { TFunction } from '../lib/functionTypes';
import { colors, fonts, mediaQueries } from '../lib/styleUtils';

interface Props {
  children: ReactNode;
  className?: string;
  onClick: TFunction;
}

const useStyles = createUseStyles({
  root: {
    alignSelf: 'center',
    background: colors.orange,
    border: 'none',
    borderRadius: 3,
    color: colors.beige,
    fontFamily: fonts.bt,
    fontSize: 32,
    padding: '8px 18px',
    textTransform: 'uppercase',
    [mediaQueries.upXs]: {
      borderRadius: 5,
      fontSize: 48,
      padding: '10px 20px'
    },
    [mediaQueries.upLg]: {
      borderRadius: 10,
      fontSize: 60,
      padding: '12px 26px'
    },

    '&:hover': {
      backgroundColor: colors.orange2
    }
  },
});

export const Button = ({ children, className, onClick }: Props) => {
  const classes = useStyles();

  return (
    <button className={classNames(classes.root, className)} onClick={onClick}>
      {children}
    </button>
  );
};
