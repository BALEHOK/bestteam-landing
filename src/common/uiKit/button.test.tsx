import React from 'react';
import { itRenders } from '../testUtils/itRenders';
import { Button, ButtonType } from './button';

describe('button', () => {
  itRenders(<Button onClick={jest.fn()}>default</Button>, 'it renders default button');
  itRenders(<Button onClick={jest.fn()} type={ButtonType.primary}>primary</Button>, 'it renders primary button');
  itRenders(<Button onClick={jest.fn()} type={ButtonType.secondary}>secondary</Button>, 'it renders secondary button');
});