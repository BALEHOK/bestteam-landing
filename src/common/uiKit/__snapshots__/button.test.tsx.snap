// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button it renders default button 1`] = `
<button
  className="root-0-2-1 primaryButtonStyle-0-2-2"
  onClick={[MockFunction]}
>
  default
</button>
`;

exports[`button it renders primary button 1`] = `
<button
  className="root-0-2-1 primaryButtonStyle-0-2-2"
  onClick={[MockFunction]}
>
  primary
</button>
`;

exports[`button it renders secondary button 1`] = `
<button
  className="root-0-2-1 secondaryButtonStyle-0-2-3"
  onClick={[MockFunction]}
>
  secondary
</button>
`;
